File: src-web/client/src/vcs/repo.ts
import { Dev } from "../common/dev";
import { UpdateHistory } from "./history";

export interface Repo {
	devs: {
		/**
		 * Have the final say and ultimate responsibility for what happens to the
		 * repo.
		 */
		lead: Dev.Email[];
		/**
		 * Members committed to the project and trusted to review code.
		 */
		core: Dev.Email[];
	};
	history: UpdateHistory;
}

File: packages/gemel/lib/src/vcs/repo.dart
import 'history.dart';

///
class Repo {
  final RepoDevs devs;
  final UpdateHistory history;

  Repo({
    required this.devs,
    required this.history,
  });
}

class RepoDevs {
  String lead;
  List<String> core;

  RepoDevs({
    required this.lead,
    required this.core,
  });
}
